app/models/question.rb

```diff
@@ -11,6 +11,20 @@
 #  updated_at                           :datetime         not null
 #
 
+class QuestionOrdersValidator < ActiveModel::Validator
+  def validate(record)
+    if record.question_type == 'short_answer'
+      if record.question_orders.length != 1
+        record.errors[:base] << 'length of question orders for short answer should be one'
+      end
+    else
+      if record.question_orders.length < 1
```

Comment:
Okay, I thought the terminating condition was that every question has a question order and it points to null. But now looking at the model generation code in your PR this is not the case.